setup.cfg

```diff
@@ -33,3 +33,6 @@ packages =
 [entry_points]
 console_scripts =
     bimmerconnected = bimmer_connected.cli:main
+
+[options.package_data]
+bimmer_connected = py.typed
```

Comment:
Not completely sure, but you might also need to set `include_package_data = True` in the `[options]` sections. Best to test it, just to make sure it's actually included.

_And then later also the release assets. It wouldn't be the first time `py.typed` wasn't included :)_

Reply:
Might be a stupid question. But how to test this? Just by running `python setup.py sdist bdist_wheel` to check if the files are there?

Or would I install the library into my HA dev environment and check it from there somehow using mypy?

Reply:
> Might be a stupid question. But how to test this? Just by running `python setup.py sdist bdist_wheel` to check if the files are there?

Exactly! That's what I do anyway since data files can be a bit tricky sometimes. It's recommended to run `python setup.py clean --all` beforehand though.

I just checked it myself since I wanted to know if `include_packge_data` is necessary. Turns out it's not with `package_data` 🤷🏻 https://setuptools.pypa.io/en/latest/userguide/datafiles.html

> Or would I install the library into my HA dev environment and check it from there somehow using mypy?

That's another good check. Install it as `--editable` and run `mypy homeassistant/components/bmw_connected_drive/`. Since the library now has a `py.typed` file it wouldn't be unexpected to see new errors pop up. The question is weather they are errors in Home Assistant or if the typing here is wrong.

Reply:
Cool! I tested it also using HA and was able to remove the last couple of issues (like 10 or so in total in only 2 files). Seems like the base wasn't that bad and I was able to keep the expected types :)

Only one thing I'm stumbling about:
- [homeassistant/components/bmw_connected_drive/sensor.py:57](https://github.com/rikroe/core/blob/bmw-async/homeassistant/components/bmw_connected_drive/sensor.py#L57): error: Argument 1 to "get" of "dict" has incompatible type "Optional[str]"; expected "str"  [arg-type]
- [homeassistant/components/bmw_connected_drive/sensor.py:57](https://github.com/rikroe/core/blob/bmw-async/homeassistant/components/bmw_connected_drive/sensor.py#L57): error: Argument 2 to "get" of "dict" has incompatible type "Optional[str]"; expected "str"  [arg-type]

The `Optional[str]` value used here is `unit.value` from [bimmer_connected/vehicle/models.py#L133](https://github.com/bimmerconnected/bimmer_connected/blob/master/bimmer_connected/vehicle/models.py#L133), which can also be `None` to allow accessing it by index even when it is empty. 
Should I just cast it as a string using `str(state.unit)` or should I rather use `# type: ignore[arg-type]` in such a case?

Reply:
The `converter` method only makes sense if the second argument is not `None`. Why not add a second condition to check if `unit` is None?
```diff
-    if state.value is None:
+    if state.value is None or state.unit is None:
         return None
     return round(
         converter(state.value, UNIT_MAP.get(state.unit, state.unit)), precision
     )
```

Reply:
Of course 😅 The HA component also is now fully typed (I'll add it to https://github.com/home-assistant/core/pull/71827 once this PR is merged and we have a published release).